### DISCUSSION TITLE
RemaFoX can't find any strings in Localizable.xcstrings

### DISCUSSION THREAD

**Discussion by noloman** [FlineDev/RemafoX#108] [Help / FAQ]
Hi,

First of all, thanks for such a great tool! I've been using it for a while and I'm very happy with it :)

I recently migrated to strings catalog with Xcode 15 and I realized that RemafoX can't find my strings and their translations anymore.
So I tried to set my project up from scratch, and even though I was using 10 languages with a lot of strings translated, now RemafoX can't find any language being used in the project:

<img width="632" alt="Screenshot 2024-02-12 at 12 57 14" src="https://github.com/FlineDev/RemafoX/assets/183264/d7bbf39c-30e8-48ef-aee3-9498391786f8">

This is the left panel of my `Localizable.xcstrings` file:

<img width="218" alt="Screenshot 2024-02-12 at 12 59 14" src="https://github.com/FlineDev/RemafoX/assets/183264/5e2cb114-93b5-4b6f-ae37-555cb2f3ff8e">

Am I doing something wrong? I followed the setup process by dragging and dropping the generated enum (`Remafox.swift)` but I see that the file is empty.

Thanks!

**Comment by Jeehut** [comment]
Hi @noloman, thank you for using RemafoX!

I actually wanted to add support for String Catalogs in RemafoX, but during the implementation, I noticed that String Catalogs actually have a fundamentally different concept than Strings(dict) files, it's not just better UI. The fundamental difference is an inversion of the source of truth: While Strings(dict) were the source of truth for your translations and I could generate an auto-completable enum file based on those, the source of truth is reversed in String Catalogs.

String Catalogs are NOT the source of truth of your translations, but your code is! Whenever you make a change in your code, the related entries in the String Catalog are marked as stale and new keys are added. That's also why Apple didn't implement generated symbols for translations like they did with Images and Colors in Xcode 15, simply because the source of truth now is your code.

So, in many ways, String Catalogs & RemafoX don't play well together. In fact, String Catalogs replace the Normalizer & Linter in RemafoX, and this reversed source of truth means that it doesn't make sense to generate code based off of the String Catalog. Your code is already the source of truth. So the only feature in RemafoX Apple didn't cover with String Catalogs is machine-translation.

So I decided for the following:
* Anybody who is already using RemafoX doesn't really need to migrate to String Catalogs if they're happy with the current workflow. There's no real advantage of String Catalogs compared to RemafoX, in fact there's RemafoX still holds the advantage of allowing for semantic keys, which is not how String Catalogs are designed.
* Anybody who wants to migrate to String Catalogs anyway can use my dedicated app [String Catalog Translator](https://apps.apple.com/app/apple-store/id6476773066?pt=549314&ct=github.com&mt=8). Unlike RemafoX, there's no setup needed which makes it way easier to use – you just drag & drop your String Catalog and it translates empty entries for you. If you have purchased RemafoX Yearly or Lifetime, just send an email to remafox@fline.dev with a screenshot of your "Plan Chooser" window where I can see your purchase and I'll give you a promo code which will give you free access to the same plan in String Catalog Translator.

This is my thoughtful answer to Apple's introduction of String Catalogs, sherlocking 70% of RemafoX.

My goal with RemafoX has always been to provide the best localization experience workflow possible. And rest assured that I'm still thinking about improving the entire workflow for developers, now based on String Catalogs. There are still areas I'm not 100% happy with which I will improve in future updates of either RemafoX or String Catalog Translator. I might also end up merging them into one app, but this seemed to be the easiest solution to start this process.

I hope this helps. Please let me know what you think.
